src/it/scala/djinni/GeneratorIntegrationTest.scala

```diff
@@ -343,6 +343,66 @@ class GeneratorIntegrationTest extends IntegrationTest with GivenWhenThen {
       assertFileContentEquals(idlFile, "", List("generated-files.txt"))
     }
 
+    it(
+      "should be able to generate C++ and C++/CLI outputs for interfaces with external dependencies and non-null pointers"
```

Comment:
I'm not 100% sure but from my understanding the support for non-null pointers that you added and the problem with external dependencies are two separate topics, that could be tested in isolation from each other? Am I correct with that? If so, can you split the test cases so only one feature is tested at a time?

Reply:
You are right. I added this test accidentally, as I wanted to add a test to test interface "dependencies" (exactly [the next test](https://github.com/cross-language-cpp/djinni-generator/pull/128/files/d9616f9dfe51483d15718327a74b02febadf5205#diff-dc8bedef9ea7fe6b135c617767cddccc31ace3e8f42f86e164d2717a63a48b74R375)), but I took the external dependencies and realized that's not what I needed.

I will split test cases and test external dependencies in one test and non-null pointers in another.

Reply:
@jothepro I split the tests:
1. C++/CLI interface with external dependencies
2. C++/CLI interface with circular dependencies
3. C++/CLI interface with non-null pointers

Could you review again the changes?

Reply:
Nice! thx!